packages/repository-json-schema/src/__tests__/integration/build-schema.integration.ts

```diff
@@ -1019,27 +1019,54 @@ describe('build-schema', () => {
 
       const expectedSchema: JsonSchema = {
         definitions: {
-          ProductWithRelations: {
-            title: 'ProductWithRelations',
+          Category: {
```

Comment:
I find it very difficult to review this part of the changeset. Is there a way how to preserve the old order of `definitions` entries, to keep the amount of changed lines as small as possible please?

Reply:
@bajtos I don't know how to proceed with this PR tbh.
The original goal was to eliminate all duplication that blows up taxonomy in OpenAPI definitions. The reason is that each model creates 4+ definitions (`Thing`, `ThingPartial`, `ThingWithRelations`, `NewThing`, etc.) and this causes issues in the client-side generated code as TypeScript does not consider all those incarnations of the same model compatible. 

This code seems to work fine on my projects but I've observed more cases where extra definitions are created. Other variations on `JsonSchemaOptions` (e.g. excluding properties) create new definitions. I'm not sure if this code is future-proof. A proper fix would be to construct an inheritance chain/tree in terms of OpenAPI schema but I don't have enough experience with lb4 to account for all scenarios. 
 
Also I'm not sure if the `isInherited` check and `getParentModel` implemented here are sufficient.

My suggestion for moving this forward would be to expose `modelToJsonSchema` to userland customization/override instead. So that we can explore the advanced scenarios and come up with patterns that work before writing them in stone.